File: console/shell/internal/status/frame.go
package status

import (
	stdstrings "strings"

	"github.com/oullin/tui/internal/environment"
	"github.com/oullin/tui/internal/terminal"
)

func RenderedFrameLineCount(frame string) int {
	trimmed := stdstrings.TrimSuffix(frame, "\n")
	if trimmed == "" {
		return 0
	}
	return len(stdstrings.Split(trimmed, "\n"))
}

func EraseRenderedFrame(frame string) {
	count := RenderedFrameLineCount(frame)
	environment.Current().Output.Write(terminal.ErasePreviousLinesSequence(count))
}

File: pkgs-ts/console/src/status/frame.ts
import { erasePreviousLines } from '#console/terminal';

export const renderedFrameLineCount = (frame: string): number => {
	if (frame.length === 0) {
		return 0;
	}

	const lines = frame.split('\n').length;

	return frame.endsWith('\n') ? lines - 1 : lines;
};

export const eraseRenderedFrame = (frame: string): void => {
	erasePreviousLines(renderedFrameLineCount(frame));
};
